src/cli/argsParser/parseArgv/mocha/parseMochaArgs.ts

```diff
@@ -0,0 +1,97 @@
+import { pick as _pick } from 'lodash'
+import Mocha from 'mocha'
+import {
+  createMissingArgumentError,
+  createUnsupportedError
+} from 'mocha/lib/errors'
+import { ONE_AND_DONES } from 'mocha/lib/cli/one-and-dones'
+import { handleRequires, validatePlugin } from 'mocha/lib/cli/run-helpers'
+import { aliases, types } from 'mocha/lib/cli/run-option-metadata'
+import yargs, { Arguments } from 'yargs'
+
+import mochaOptions from './mochaOptions'
+import { ParsedMochaArgs } from './types'
+
+// Used to establish parity with Mocha's builder function in lib/cli/run.js
+const mochaChecks = (yargsInstance: any, argv: Arguments) => {
+  // "one-and-dones"; let yargs handle help and version
+  Object.keys(ONE_AND_DONES).forEach(opt => {
+    if (argv[opt]) {
+      ONE_AND_DONES[opt].call(null, yargsInstance)
+      process.exit()
+    }
+  })
+
+  // yargs.implies() isn't flexible enough to handle this
+  if (argv.invert && !('fgrep' in argv || 'grep' in argv)) {
+    throw createMissingArgumentError(
+      '"--invert" requires one of "--fgrep <str>" or "--grep <regexp>"',
+      '--fgrep|--grep',
+      'string|regexp'
+    )
+  }
+
+  if (argv.compilers) {
+    throw createUnsupportedError(
+      `--compilers is DEPRECATED and no longer supported.
+      See https://git.io/vdcSr for migration information.`
+    )
+  }
+
+  /**
+   * Commented out here to provide backward compatibility for now
+   */
+  /*
+  if (argv.opts) {
+    throw createUnsupportedError(
+      `--opts: configuring Mocha via 'mocha.opts' is DEPRECATED and no longer supported.
+      Please use a configuration file instead.`
+    )
+  }
+  */
+
+  // load requires first, because it can impact "plugin" validation
+  handleRequires(argv.require)
```

Comment:
@Jack-Barry Is is intended that we invoke `requires` BEFORE webpack compilation? It seems strange. The intention of the user is to use require for global setup between all tests run, but not before webpack compilation, after it. For example if user requires `jsdom-global` it will setup browser-like environment and will break compilation, because webpack loaders will think they are invoked in the browser. And the user intent was to invoke `jsdom-global` before all tests to provide the browser-environment for the tests.

Reply:
Disregard, there is `--include` for that